### PR TITLE
Simplify/fix windowing check in BeamSortRel

### PR DESCRIPTION
**Pull request by kennknowles**
This is a small piece inspired by #5805. Before this, `BeamSortRel` rejected LIMIT on non-global windows.

As noted in #5805, it doesn't really make sense to keep the windowing strategy around on PCollections. So since the PCollections representing rels are conceptually globally windowed, it is safe for LIMIT to re-window into the global window in order to stream the desired number of rows.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @apilloud @amaliujia

**Comment by amaliujia** [comment]
My concern is if and when we will support setting windowing strategy in the further, it could take some time to debug for other people because of setting a fixed global window for LIMIT in code.

It would be debatable that if per window LIMIT makes sense though. I believe per window LIMIT is useful.

**Comment by amaliujia** [comment]
LGTM

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java [hunk 1]
```diff
@@ -154,22 +155,29 @@ public int getCount() {
...
 
-      BeamSqlRowComparator comparator =
-          new BeamSqlRowComparator(fieldIndices, orientation, nullsFirst);
+      // There is a need to separate ORDER BY LIMIT and LIMIT:
+      //  - GroupByKey (used in Top) is not allowed on unbounded data in global window so ORDER BY ... LIMIT
+      //    works only on bounded data.
+      //  - Just LIMIT operates on unbounded data, but across windows.
+      if (fieldIndices.size() == 0) {
+        return upstream
+            .apply(Window.into(new GlobalWindows()))
```
Comment (amaliujia) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java]: Nit: let's have a TODO here to remind that we might need this fixed global window strategy to a user defined one?

**Comment by kennknowles** [comment]
I agree with your points. I think per-window limit makes sense, I just don't know how to express it in SQL query. The window is "just another key" in the GROUP BY so it is like this question: https://stackoverflow.com/questions/2129693/using-limit-within-group-by-to-get-n-results-per-group

My actual view is this: either the query is per-window because it is inline in a pipeline (transforms are expected to automatically operate per-window) or all rels are in the global window and windowing exists only during a GROUP BY (BeamAggregationRel).

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java (hunk 1 above)
Comment (kennknowles) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java]: Done

**Pull request merged**